traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go

```diff
@@ -266,6 +266,14 @@ func createV30(w http.ResponseWriter, r *http.Request, inf *api.APIInfo, ds tc.D
 		deepCachingType = ds.DeepCachingType.String() // necessary, because DeepCachingType's default needs to insert the string, not "", and Query doesn't call .String().
 	}
 
+	if ds.Topology != nil {
```

Comment:
These checks seem like they belong in the `Validate` chain rather than here.

Reply:
The Validate chain doesn't allow reporting system errors, so that first error check would be swallowed.

Reply:
Well, well, well... should we add a `// TODO: ...` for that? And in the meantime, can we refactor the check into a shared function that is used in create() as well as update()?

Reply:
> ...in the meantime, can we refactor the check into a shared function that is used in create() as well as update()?

Yeah, for sure.

> ...  should we add a // TODO: ... for that?

It'll mean changing the `github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api.ParseValidator` interface, but I think that'd be a worthwhile change. So: yes, and that's where I'll put it.